stage #2.5/module #3 Spring Core/#5.2. Classpath Scanning and Managed Components/content.md

```diff
@@ -0,0 +1,238 @@
+
+The following example shows the configuration ignoring all `@Repository` annotations and using "stub" repositories
+instead ([see more details about filtering options][2]):
+
+```java
+
+@Configuration
+@ComponentScan(basePackages = "com.somepackage",
+        includeFilters = @Filter(type = FilterType.REGEX, pattern = ".*Stub.*Repository"),
+        excludeFilters = @Filter(Repository.class))
+public class Config {
+    // ...
+}
+```
+
+## Defining Bean Metadata within Components
+
+Spring components can also contribute bean definition metadata to the container. You can do this with the same `@Bean`
+annotation used to define bean metadata within `@Configuration` annotated classes. The following example shows how to do
+so:
+
+```java
+
+@Component
+public class MyComponent {
+
+    @Bean
+    @Qualifier("main")
+    public MyService myService() {
+        return new MyServiceImpl();
+    }
+
+    @Bean
+    protected MyAnotherService myAnotherService(@Qualifier("main") MyService myService) {
+        return new MyAnotherServiceImpl(myService);
+    }
+
+    public void doWork() {
+        // ...
+    }
+
+    // ...
+}
+```
+
+The preceding class is a Spring component that has application-specific code in its `doWork()` method. However, it also
+contributes bean definitions that have factory methods referring to the methods `myService()` and `myAnotherService()`.
+The `@Bean` annotation identifies the factory methods and other bean definition properties, such as a qualifier value
+through the `@Qualifier`annotation. Other method-level annotations that can be specified are `@Scope`, `@Lazy`, and
+custom qualifier annotations. Autowired fields and methods are also supported with additional support for autowiring
+of `@Bean` methods.
+
+## Naming Autodetected Components
+
+When a component is autodetected as part of the scanning process, its bean name is generated by the `BeanNameGenerator`
+strategy known to that scanner. By default, any Spring stereotype annotation (`@Component`, `@Repository`,
+`@Service`, and `@Controller`) that contains a name `value` thereby provides that name to the corresponding bean
+definition.
+
+If such an annotation contains no name `value` or for any other detected component (such as those discovered by custom
+filters), the default bean name generator returns the uncapitalized non-qualified class name. For example, if the
+following component classes were detected, the names would be `myMusicService` and `musicRatesFinderImpl`:
+
+```java
+
+@Service("myMusicService")
+public class MusicService {
+    // ...
+}
+
+@Repository
+public class MusicRatesFinderImpl implements MusicRatesFinder {
+    // ...
+}
+```
+
+If you do not want to rely on the default bean-naming strategy, you can provide a custom bean-naming strategy. First,
+implement the `BeanNameGenerator` interface, and be sure to include a default no-arg constructor. Then, provide the
```

Comment:
I believe no need to tell how to create custom bean-naming strategy. There are already exhaustive amount of info in this chapter.

Reply:
Yes, I agree that there is already an exhaustive amount of info, but I believe in context of the corresponding section it worth to mention and say some words that user can customize bean naming strategy.

Reply:
Ok, leave it